src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -264,20 +277,47 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 		return exprStmt.WithExpression( result );
 	}
 
-	private InvocationExpressionSyntax Transform( InvocationExpressionSyntax invocationExpr) {
+	private ExpressionSyntax Transform( InvocationExpressionSyntax invocationExpr) {
+		ExpressionSyntax newExpr = invocationExpr;
 		var memberAccess = invocationExpr.Expression as MemberAccessExpressionSyntax;
+
 		if( string.Equals( memberAccess?.Name?.Identifier.ValueText, "ConfigureAwait", StringComparison.Ordinal ) ) {
 			if( memberAccess?.Expression is not null ) {
-				invocationExpr = (InvocationExpressionSyntax)memberAccess.Expression;
+				newExpr = memberAccess.Expression;
+				return Transform( newExpr );
 			}
 		}
+		else if( memberAccess is not null && MemberAccessesToWrapInTaskRun( memberAccess ) ) {
+			disableTaskRunWarningFlag = true;
+			return SyntaxFactory.ParseExpression( $"Task.Run(() => {invocationExpr}).Result" );
+		}
+
 		return invocationExpr
 			.WithExpression( Transform( invocationExpr.Expression ) )
 			.WithArgumentList( TransformAll( invocationExpr.ArgumentList, Transform ) );
 	}
 
-	private MemberAccessExpressionSyntax Transform( MemberAccessExpressionSyntax memberAccessExpr ) {
+	bool ReturnedMemberAccessesToRemove( MemberAccessExpressionSyntax memberAccessExpr )
+	  => memberAccessExpr.Name.Identifier.ValueText switch {
+		  "FromResult" => true,
+		  "CompletedTask" => true,
+		  _ => false
+	  };
+
+	bool MemberAccessesToWrapInTaskRun( MemberAccessExpressionSyntax memberAccessExpr )
+	  => memberAccessExpr.Name.Identifier.ValueText switch {
+		  "ReadAsStringAsync" => true,

```

Comment:
We might eventually want to be a bit more careful with this stuff, like also look at the type on the left side to make sure we're talking about the right ReadAsStringAsync etc. and not some other classes method, but I'm OK with this for now.

Reply:
I made a change to look on the left side as well. I just set the left side that it could be anything for ReadAsStringAsync for now though since I'm looking at name and not type at the moment (there's res.Content.ReadAsStringAsync, response.Content.ReadAsStringAsync, etc.)